syntax = "proto3";

package scheduler;

// defines the days and hours worked
message Shift {
    DayOfTheWeek day_of_the_week = 1;
    HourOfTheDay start_hour = 2;
    HourOfTheDay end_hour = 3;
    Employee employee = 4;
}

// defines values for days of the week
enum DayOfTheWeek {
    UKNOWN = 0;
    MONDAY = 1;
    TUESDAY = 2;
    WEDNESDAY = 3;
    THURSDAY = 4;
    FRIDAY = 5;
    SATURDAY = 6;
    SUNDAY = 7;
}

// defines hours of the day
enum HourOfTheDay {
    UNKNOWN = 0;
    ONE = 1;
    TWO = 2;
    THREE = 3;
    FOUR = 4;
    FIVE = 5;
    SIX = 6;
    SEVEN = 7;
    EIGHT = 8;
    NINE = 9;
    TEN = 10;
    ELEVEN = 11;
    TWELVE = 12;
    THIRTEEN = 13;
    FOURTEEN = 14;
    FIFTEEN = 15;
    SIXTEEN = 16;
    SEVENTEEN = 17;
    EIGHTEEN = 18;
    NINETEEN = 19;
    TWENTY = 20;
    TWENTYONE = 21;
    TWENTYTWO = 22;
    TWENTYTHREE = 23;
    TWENTYFOUR = 24;
}

// defines the employee and hours/shifts worked
message Employee {
    string employee_id = 1;
    int32 base_pay = 2;
    repeated ShiftList shift_list = 3;
    int32 weekend_bonus = 4;
    int32 night_bonus = 5;
}

message ShiftList {
    string shift_name = 1;
}